lib/gooddata/models/process.rb

```diff
@@ -90,16 +90,23 @@ def upload_package(path, files_to_exclude, opts = { :client => GoodData.connecti
       # @option options [String] :name Readable name of the process
       # @option options [String] :process_id ID of a process to be redeployed (do not set if you want to create a new process)
       # @option options [Boolean] :verbose (false) Switch on verbose mode for detailed logging
-      def deploy(path, options = { :client => GoodData.client, :project => GoodData.project })
-        return deploy_brick(path, options) if path.to_s.start_with?(APP_STORE_URL)
-
-        return deploy_from_appstore(path.to_s, options) if (path.to_s =~ %r{\${.*}:(.*)\/(.*):\/}) == 0 # rubocop:disable Style/NumericPredicate
+      def deploy(path, options = { client: GoodData.client, project: GoodData.project })
+        if (path.is_a?(Hash) && path[:component])
```

Comment:
Style/ParenthesesAroundCondition: Don't use parentheses around the condition of an `if`.